### ISSUE TITLE
Type alias'ed builtin types are not being expanded

### ISSUE THREAD

**Issue by juanpablos** [microsoft/pylance-release#315]
## Environment data
-   Language Server version: 2020.8.3
-   OS and version: Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.8.5

## Expected behaviour

Type hints for the generic part of some builtin generic types. The comments indicate the type reported by Pylance.
Pylance 2020.8.2 and older display `List[int]` for variables annotated with `List[int]`.
```python
l1 = []  # (variable) list[Any]
l2 = [i for i in range(10)]  # (variable) list[int]
l3: List[int] = []  # (variable) list[int]
l4: List[int] = [i for i in range(10)]  # (variable) list[int]
```

## Actual behaviour

Since Pylance 2020.8.3, variables annotated with `List[int]` are displayed as `(type alias) List` instead of the probably intended `List[int]`.
```python
l1 = []  # (variable) list[Any]
l2 = [i for i in range(10)]  # (variable) list[int]
l3: List[int] = []  # (type alias) List
l4: List[int] = [i for i in range(10)]  # (type alias) List
```

## Code Snippet / Additional information

The problem is only present with builtin types that are defined as `... = _Alias()` in the `typing` module. For other generic types that are defined as classes like `Iterable`, or custom generic types created using `Generic`, the type hinting works fine.
This might be a side effect of #265 because `List`, for instance, is defined as `List = _Alias()` in the `typing` module.
A workaround for now is to import`annotations` from `__future__`, but doing so in every module might be too much.

**Comment by erictraut** [comment]
This behavior is intended. If a type annotation of `List[int]` is provided for a variable, most users would expect to see this in the hover text, not some other type. We try to preserve type aliases in hover text and diagnostic messages wherever possible. 

I presume that your concern is purely one of aesthetics or personal preference? Or is there some practical argument for not using the typing aliases in this context? I can't think of any ambiguity or confusion that would be caused by displaying the typing class versus the builtin class.

**Comment by jakebailey** [comment]
I do think there's a bug here, though, which I've seen in other contexts. Hovering over `l4` shouldn't claim that the variable itself is a type alias. I've seen this for global variables before where it claims "type alias" when it isn't; maybe an issue in the type alias heuristic.

![image](https://user-images.githubusercontent.com/5341706/92048434-8dbcc200-ed3c-11ea-998f-b00318238441.png)

**Comment by jakebailey** [comment]
E.g., why isn't it saying `List[int]`?

**Comment by jakebailey** [comment]
The global-ness doesn't seem to matter, actually.

![image](https://user-images.githubusercontent.com/5341706/92048618-f99f2a80-ed3c-11ea-8f27-b239948f5155.png)

**Comment by erictraut** [comment]
@jakebailey, yes, that is a bug in the hover provider. It shouldn't display "type alias" in this case. It should just say "variable".

**Comment by erictraut** [comment]
Maybe we should file a separate bug (in pyright) to track that issue. It's different from the issue that was reported above.

**Comment by jakebailey** [comment]
Is it? What I'm describing is what I'm reading in the original report, that they expected to see variable in the hovers, but we said type alias (comparing expected to actual, they are copy-pastes from hovers).

**Comment by jakebailey** [comment]
e.g., we'd output:

```python
l1 = []  # (variable) list[Any]
l2 = [i for i in range(10)]  # (variable) list[int]
l3: List[int] = []  # (variable) List[int]
l4: List[int] = [i for i in range(10)]  # (variable) List[int]
```

Unless you are specifically talking about `List` versus `list`.

**Comment by erictraut** [comment]
Ah, you're right. I missed that. OK, I'll dig into it.

**Comment by juanpablos** [comment]
Sorry should have worded it better. I just copy-pasted the hovers and put them as comments. Thanks for the follow up!

**Comment by erictraut** [comment]
I've fixed a couple of bugs that caused incorrect identification of "type alias" in the hover text, and changed the inference logic to use "List", "Set" and "Dict" rather than "list", "set" and "dict" when inferring the type of a list, set or dict expression. These are equivalent, but the upper-case versions provide consistency with type annotations.

**Comment by huguesv** [comment]
This issue has been fixed in version 2020.9.4, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#202094-10-september-2020
